packages/affiliates/libs/uniswap/contracts.ts

```diff
@@ -0,0 +1,135 @@
+import assert from "assert";
+import { ethers } from "ethers";
+import Promise from "bluebird";
+
+import UniswapV3Pool from "@uniswap/v3-core/artifacts/contracts/UniswapV3Pool.sol/UniswapV3Pool.json";
+
+import { Positions, Position, Pool } from "./models";
+import { PoolEvents } from "./processor";
+import { convertValuesToString } from "./utils";
+
+type NetworkName = "kovan" | "rinkeby" | "ropsten";
+type ContractType =
+  | "v3CoreFactoryAddress"
+  | "weth9Address"
+  | "multicall2Address"
+  | "proxyAdminAddress"
+  | "tickLensAddress"
+  | "quoterAddress"
+  | "swapRouter"
+  | "nonfungibleTokenPositionDescriptorAddress"
+  | "descriptorProxyAddress"
+  | "nonfungibleTokenPositionManagerAddress"
+  | "v3MigratorAddress";
+
+type NetworkContracts = {
+  [key in ContractType]: string;
+};
+
+// Maybe this is available somewhere in uniswaps module, but not sure
+const networks = new Map<NetworkName, NetworkContracts>([
```

Comment:
Mainly just curious (nothing wrong with what you did here), but why did you choose to use a map rather than a normal object?
```ts
type Networks = {
  [key in NetworkName]: {
    [key in ContractType]: string
  }
};

const networks: Networks = {
  kovan: {
    v3CoreFactoryAddress: "0x7046f9311663DB8B7cf218BC7B6F3f17B0Ea1047",
    weth9Address: "0xd0A1E359811322d97991E03f863a0C30C2cF029C",
    ...
  },
  ...
};
```

Note: this _may_ (not sure) give you a guarantee that all lookups into this object return values that are well-defined (not null or undefined), so you won't need to assert it.

Reply:
i tend to use maps in JS because i can check for existince with `has`, and it will correctly return true for values like 0,, which you have to explicitly check for with an object. With typescript though I think you are on to something, I can go back to objects and have strong guarantees and more concise code.